library/pem.c

```diff
@@ -423,9 +423,11 @@ int mbedtls_pem_read_buffer( mbedtls_pem_context *ctx, const char *header, const
 
 void mbedtls_pem_free( mbedtls_pem_context *ctx )
 {
-    if( ctx->buf != NULL )
+    if ( ctx->buf != NULL )
+    {
         mbedtls_platform_zeroize( ctx->buf, ctx->buflen );
-    mbedtls_free( ctx->buf );
+        mbedtls_free( ctx->buf );
```

Comment:
As mentioned in [this post](https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c):
>Quoting the C standard, 7.20.3.2/2 from [ISO-IEC 9899](http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf):
`void free(void *ptr);`
If `ptr` is a null pointer, no action occurs.
Don't check for `NULL`, it only adds more dummy code to read and is thus a bad practice.

I don't see an added value in this change

Reply:
The quoted post does not apply here. No huge gain herre, but no dummy code.
Check for null pointer was necessary to call zeroize function safely.

Full quotation from the standards:
> The free function causes the space pointed to by ptr to be deallocated, that is, made available for further allocation. If ptr is a null pointer, no action occurs.

`No action` here means that no attemt to derefence null pointer would be tried. But argument pointer had to be passed, function called, argument checked for null, and then return from the function. All these _actions_ could be skipped if it was already known that the pointer was null. 
Very smart compiler might inline the call, and then optimize, but two added braces do this always and without increase in code size.

Reply:
As we have the check for `ctx->buf != NULL` anyway, I don't mind moving the call to `mbedtls_free()` here, although the benefit seems somewhere between marginal and unnoticeable.